src/main/java/uk/gov/hmcts/reform/iacaseaccessapi/infrastructure/service/CcdSupplementaryDetailsSearchService.java

```diff
@@ -63,14 +73,31 @@ public CcdSupplementaryDetailsSearchService(SystemTokenGenerator systemTokenGene
             throw new IdentityManagerResponseException(e.getMessage(), e);
         }
 
-        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
-        TermsQueryBuilder termQueryBuilder = QueryBuilders.termsQuery("reference", ccdCaseNumberList);
-        searchSourceBuilder.size(100);
-        searchSourceBuilder.from(0);
-        searchSourceBuilder.sort("created_date", SortOrder.DESC);
-        searchSourceBuilder.query(termQueryBuilder);
+        List<SupplementaryInfo> allResults = new ArrayList<>();
+        final Collection<List<String>> chunkedCcdCaseList = getChunkedCcdCaseList(ccdCaseNumberList);
+        final int proposedPoolSize = (int) Math.ceil((double) ccdCaseNumberList.size() / maxRecords);
+        ExecutorService executorService = Executors.newFixedThreadPool(proposedPoolSize);
```

Comment:
`ExecutorService` could be injected in contructor as a Bean, that will safe time to create a new pool in every request
It is safe to specify number of threads in configuration. I guess 10 threads is enough and ok to start with, you also don't need to calculate it each time. You can have `application.yaml` property and `ExecutoreService` created in one of the spring configuration files